src/import_whatsapp.js

```diff
@@ -235,57 +277,89 @@ const processText = (text) => {
 			}
 		}
 	});
+	return messages;
+};
+
+const processText = (text) => {
+	const groupNameRegex = /"([^"]*)"/;
+	const groupNameMatches = text.match(groupNameRegex);
+	const groupName = groupNameMatches ? groupNameMatches[1] : null;
+
+	// Check the first 3 characters to determine the format; iOS and Android
+	const fileType = text.substring(0, 3);
+	const [messageRegex, imgFileRegex] = setImgMsgRegex(fileType);
+
+	let messageMatches = [...text.matchAll(messageRegex)];
+
+	// Convert messageMatches to array of JSON objects and then sort
+	let [messages, senders] = processMsgMatches(messageMatches, imgFileRegex);
+	messages = sortMessages(messages);
 
 	// Now loop through messages to create geojson for each location
 	var mapdata = {
 		type: "FeatureCollection",
 		features: [],
 	};
-	let feature = null;
+	let currentFeature = null;
 	let currentSender = null;
 
+	const createFeature = (message, groupName) => {
+		return {
+			type: "Feature",
+			properties: {
+				contributionid: crypto.randomUUID(),
+				mainattribute: groupName,
+				observations: "",
+				observer: message.sender,
+				datetime: message.datetime,
+				markerColour: senders[message.sender],
+				imgFilenames: [],
+			},
+			geometry: message.location
+				? {
+					type: "Point",
+					coordinates: [message.location.long, message.location.lat],
+				}
+				: null,
+		};
+	};
+
+	const isValidContent = (content) => {
```

Comment:
The name of this function suggests it's going to return true or false, but it seems to be doing more cleaning? Do we need this or could it be wrapped up into cleanMsgContent? Some comments describing the purpose would be useful.

Reply:
originally added this to try removing the "remove_this_msg" and it didn't work so I think you're right that it can just be bundled into the cleaning function